From fbe8c547de05d84c9244f21b4049efe78ebc3dc6 Mon Sep 17 00:00:00 2001
Message-Id: <fbe8c547de05d84c9244f21b4049efe78ebc3dc6.1436312509.git.chang-joon.lee@intel.com>
In-Reply-To: <d7f55a0f6423e9b14861f607af19ccac4e7f31d5.1436312509.git.chang-joon.lee@intel.com>
References: <d7f55a0f6423e9b14861f607af19ccac4e7f31d5.1436312509.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Tue, 23 Jun 2015 17:35:00 +0530
Subject: [PATCH 02/15] SQUASHME! [VPG]: drm/i915: Match mutex lock & unlock in
 gem read objects.

i915_gem_read_objects missed a mutex unlocking leading to a deadlock.

SQUASHME! - This patch should be squashed into the following existing patch:
Author: Sourab Gupta <sourab.gupta@intel.com>
Date:   Mon Aug 11 13:12:04 2014 +0530
Change-Id: I02552d54a753fcfb94b5bdd9bc9c82702bbf514d
MUST_REBASE [VPG]: drm/i915: Sysfs interface to get GFX shmem usage stats per process

Issue: GMINL-11890
Change-Id: I22b7131939cd7831cffb75550e73d8a3b6018a7a
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_sysfs.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_sysfs.c b/drivers/gpu/drm/i915/i915_sysfs.c
index 9e6f45e..6898519 100644
--- a/drivers/gpu/drm/i915/i915_sysfs.c
+++ b/drivers/gpu/drm/i915/i915_sysfs.c
@@ -1325,9 +1325,11 @@ static ssize_t i915_gem_read_objects(struct file *filp,
 	}
 
 	if (!attr || !attr->private) {
+		ret = -EINVAL;
 		DRM_ERROR("attr | attr->private pointer is NULL\n");
-		return -EINVAL;
+		goto out;
 	}
+
 	attr_priv = attr->private;
 	tgid = attr_priv->tgid;
 
-- 
1.9.1

